tests/unit/mainwindow/test_prompt.py

```diff
@@ -130,3 +131,54 @@ def test_root_path(self, get_prompt):
         """With / as path, show root contents."""
         prompt = get_prompt('/')
         assert prompt._file_model.rootPath() == '/'
+
+    @pytest.fixture
+    def home_path(self, monkeypatch, tmp_path):
+        homedir = str(tmp_path)
+        monkeypatch.setenv('HOME', homedir)  # POSIX
+        monkeypatch.setenv('USERPROFILE', homedir)  # Windows
+        assert str(pathlib.Path.home()) == homedir
+        return tmp_path
+
+    def test_expand_user(self, qtbot, get_prompt, home_path):
+        for directory in ['bar', 'bat', 'foo']:
+            (home_path / directory).mkdir()
+
+        prompt = get_prompt('/')
+        prompt._lineedit.setText('')
+        with qtbot.waitSignal(prompt._file_model.directoryLoaded):
+            # _set_fileview_root isn't run unless there is an actual keypress
+            qtbot.keyPress(prompt._lineedit, Qt.Key_AsciiTilde)
```

Comment:
Only the `keyPress` line should be in the `with` block, since that's (hopefully) what's causing the signal to happen. The rest of the test should then be outside of it.

(same for the other tests)